### PR TITLE
Various accessibility API updates.

### PR DESCRIPTION
**Pull request by ndarilek**
# Objective

`bevy_a11y` was impossible to integrate into some third-party projects in part because it insisted on managing the accessibility tree on its own.

## Solution

The changes in this PR were necessary to get `bevy_egui` working with Bevy's AccessKit integration. They were tested on a fork of 0.11, developed against `bevy_egui`, then ported to main and tested against the `ui` example.

### Changed

* Add `bevy_a11y::ManageAccessibilityUpdates` to indicate whether the ECS should manage accessibility tree updates.
* Add getter/setter to `bevy_a11y::AccessibilityRequested`.
* Add `bevy_a11y::AccessibilitySystem` `SystemSet` for ordering relative to accessibility tree updates.
* Upgrade `accesskit` to v0.12.0.

### Fixed

* Correctly set initial accessibility focus to new windows on creation.

#### Before

```
use std::sync::atomic::Ordering;

// To access
accessibility_requested.load(Ordering::SeqCst)
// To update
accessibility_requested.store(true, Ordering::SeqCst);
```

#### After

```
// To access
accessibility_requested.get()
// To update
accessibility_requested.set(true);
```

### CODE REVIEW

**Comment by viridia** [comment]
One of the checks failed, looks like a problem with the Cargo.lock file, has a duplicate entry. Since the PR doesn't include Cargo.lock you be able to re-run the tests or rebase.

**Comment by ndarilek** [comment]
Interesting, thanks, any idea how a duplicate entry got in there? Didn't even know that was possible.

File: crates/bevy_a11y/src/lib.rs [hunk 1]
```diff
@@ -47,29 +88,24 @@ impl From<NodeBuilder> for AccessibilityNode {
...
-impl AccessKitEntityExt for Entity {
-    fn to_node_id(&self) -> NodeId {
-        let id = NonZeroU128::new(self.to_bits() as u128 + 1);
-        NodeId(id.unwrap())
-    }
-}
-
 /// Resource representing which entity has keyboard focus, if any.
 #[derive(Resource, Default, Deref, DerefMut)]
 pub struct Focus(Option<Entity>);
```
Comment (StaffEngineer) [inline crates/bevy_a11y/src/lib.rs:93]: pub struct Focus(pub Option<Entity>);

Comment (StaffEngineer) [inline crates/bevy_a11y/src/lib.rs:93]:
```suggestion
pub struct Focus(pub Option<Entity>);
```

File: crates/bevy_winit/src/accessibility.rs [hunk 2]
```diff
@@ -99,7 +81,7 @@ fn update_accessibility_nodes(
     )>,
     node_entities: Query<Entity, With<AccessibilityNode>>,
 ) {
-    if !accessibility_requested.load(Ordering::SeqCst) {
+    if !accessibility_requested.get() || !manage_accessibility_updates.get() {
```
Comment (alice-i-cecile) [inline crates/bevy_winit/src/accessibility.rs:84]: This will be slightly more efficient (and clearer) if we split this out into a run condition.

**Review by alice-i-cecile — APPROVED** [review]
Seems fine :) Suggestions are non-blocking.

File: crates/bevy_a11y/src/lib.rs (hunk 1 above)
Comment (ndarilek) [reply crates/bevy_a11y/src/lib.rs:93]: Oh wow, great catch. Thanks~!

File: crates/bevy_winit/src/accessibility.rs (hunk 2 above)
Comment (ndarilek) [reply crates/bevy_winit/src/accessibility.rs]: Thanks, good suggestion. Fixed.

**Comment by ndarilek** [comment]
I'm not clear where the conflict is, I merged main and everything went fine. Is this a github bug or am I genuinely missing something?

**Comment by ndarilek** [comment]
OK, maybe that got it, just spotted the link to resolve.

**Review by geom3trik — APPROVED** [review]

**Review by StaffEngineer — APPROVED** [review]

**Pull request merged**